openedx/core/djangoapps/call_stack_manager/tests/tests.py

```diff
@@ -0,0 +1,162 @@
+"""
+Test cases for Call Stack Manager
+"""
+from .models import ModelMixin, ModelNothing, ModelMixinCSM, ModelAnotherCSM
+from django.test import TestCase
+from testfixtures import LogCapture
+from openedx.core.djangoapps.call_stack_manager import donottrack
+
+
+class TestingCallStackManager(TestCase):
+    """
+    Tests for call_stack_manager package
+    """
+    def setUp(self):
+        super(TestingCallStackManager, self).setUp()
+
+    def tearDown(self):
+        """ Deleting all databases after tests
+        """
+        super(TestingCallStackManager, self).tearDown()
+        ModelMixin.objects.all().delete()
+        ModelMixinCSM.objects.all().delete()
+        ModelNothing.objects.all().delete()
+        ModelAnotherCSM.objects.all().delete()
+
+    def test_save(self):
+        """ tests save functionality of call stack manager
+        1. classes without CallStackMixin should not participate in logging.
+        """
+        with LogCapture() as l:
+            modelMixin_obj1 = ModelMixin(id_field=1, text_field="modelmixin1", float_field=12.34)
+            modelMixin_obj1.save()
+
+            modelNothing_obj1 = ModelNothing(id_field=1, name_field="ModelNothing")
+            modelNothing_obj1.save()
+
+            # Example - logging new call stack for openedx.core.djangoapps.call_stack_manager.tests.models.modelmixin
+            latest_log = l.records[-1].getMessage()[:l.records[-1].getMessage().find(':')]
+            latest_class = latest_log[latest_log.rfind('.') + 1:]
+
+            # desired latest class here
+            desired_class = "ModelMixin"
+
+            self.assertEqual(latest_class, desired_class, msg="Latest logged event should belong to " + desired_class)
```

Comment:
It would be good to test separately that the model with the mixin logs, and that the model without the mixin doesn't.